src/config.rs

```diff
@@ -89,6 +91,10 @@ impl<'a> OverrideCfg<'a> {
             },
             components: file.toolchain.components.unwrap_or_default(),
             targets: file.toolchain.targets.unwrap_or_default(),
+            profile: match file.toolchain.profile {
+                Some(name) => Some(dist::Profile::from_str(&name)?),
+                None => None,
+            },
```

Comment:
You can could use `Option::map` here, right?

Reply:
Mmm, a combination of `.map()` and `.transpose()` might make that work.

Reply:
Yeah, I was thinking of `and_then`, not `map`.